src/Checkbox/Checkbox.js

```diff
@@ -13,18 +13,28 @@ export const styleSheet = createStyleSheet('Checkbox', (theme) => {
     checked: {
       color: theme.palette.accent[500],
     },
+    label: {
+      marginLeft: -12,
+      display: 'flex',
+      alignItems: 'center',
+      cursor: 'pointer',
+    },
   };
 });
 
 export default function Checkbox(props, context) {
-  const { className, checkedClassName, ...other } = props;
+  const { className, checkedClassName, label, labelClassName, ...other } = props;
   const classes = context.styleManager.render(styleSheet);
   return (
-    <SwitchBase
-      className={classNames(classes.default, className)}
-      checkedClassName={classNames(classes.checked, checkedClassName)}
-      {...other}
-    />
+    <label className={classNames(classes.label, labelClassName)} role="presentation">
```

Comment:
That label logic is very close to the `Radio` one. I believe that it would be the same one.
What if we were implementing it using an HoC?
```jsx
const formLabel  = Component => Component;
```
We might need to use `https://github.com/mridgway/hoist-non-react-statics`.
Or we could use a simple internal `FormLabel` component.